### PR TITLE
Class Decoration : OOP School Library

### PR DESCRIPTION
**Pull request by ehmaddd**
In this PR, the following features are applied :

**Interface**
* Created a class Nameable.
- Implemented a method called correct_name that will raise a NotImplementedError.
- Turned Person class to Nameable
- Person class inherits from Nameable
- This class has a method correct_name implemented which returns the name attribute.

----------------------------------------------------------------------------------------------

**Prepared base Decorator**

- It inherits from Nameable.
- In the constructor, assigned a nameable object from params to an instance variable.
- Implemented the correct_name method that returns the result of the correct_name method of the @nameable.

----------------------------------------------------------------------------------------------

**Prepared CapitalizeDecorator**

- Created a class that inherits from the base Decorator class.
- Implemented a method correct_name that capitalizes the output of @nameable.correct_name.

----------------------------------------------------------------------------------------------

**Prepared TrimmerDecorator**
- Created a class that inherits from the base Decorator class.
- Implemented a method correct_name that makes sure the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it trims the word.

----------------------------------------------------------------------------------------------

FINALLY CHECKED THE CODE FOR WORKING AND HERE IS THE OUTPUT : 
image

### CODE REVIEW

File: trimmer_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'decorator'
+
+module TrimmerDecoratorClass
+  include DecoratorClass
+  class TrimmerDecorator < Decorator
+    def correct_name
+      return @component.correct_name.slice(0, 9) if @component.correct_name.length > 10
+
+      @component.correct_name
+    end
```
Comment (abdulhamiid) [inline trimmer_decorator.rb]: For the `TrimmerDecorator`, 

According to the [project requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md), the method `correct_name` should trim longer words to a  maximum of 10 characters, not 9. 🤓

tempsnip

Kindly fix that 👍

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @ehmaddd 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! :running_man: :runner: 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Implement composition 
- ✔️ Run a program using the command line.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: trimmer_decorator.rb (hunk 1 above)
Comment (ehmaddd) [reply trimmer_decorator.rb]: Fixed as per instructions pl'

**Review by joel-kalema — APPROVED** [review]
Hi @ehmaddd 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 


## Highlights
Linters are passing 👍 
The project is running successfully 👍 
PR is well descriptive 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**